runtime/parachains/src/hrmp/benchmarking.rs

```diff
@@ -0,0 +1,294 @@
+		register_parachain_with_balance::<T>(sender_id, deposit);
+		register_parachain_with_balance::<T>(recipient_id, deposit);
+
+		let capacity = Configuration::<T>::config().hrmp_channel_max_capacity;
+		let message_size = Configuration::<T>::config().hrmp_channel_max_message_size;
+	}: _(sender_origin, recipient_id, capacity, message_size)
+	verify {
+		assert_last_event::<T>(
+			Event::<T>::OpenChannelRequested(sender_id, recipient_id, capacity, message_size).into()
+		);
+	}
+
+	hrmp_accept_open_channel {
+		let [(sender, _), (recipient, recipient_origin)] =
+			establish_para_connection::<T>(1, 2, ParachainSetupStep::Requested);
+	}: _(recipient_origin, sender)
+	verify {
+		assert_last_event::<T>(Event::<T>::OpenChannelAccepted(sender, recipient).into());
+	}
+
+	hrmp_close_channel {
+		let [(sender, sender_origin), (recipient, _)] =
+			establish_para_connection::<T>(1, 2, ParachainSetupStep::Established);
+		let channel_id = HrmpChannelId { sender, recipient };
+	}: _(sender_origin, channel_id.clone())
+	verify {
+		assert_last_event::<T>(Event::<T>::ChannelClosed(sender, channel_id).into());
+	}
+
+	// NOTE: a single parachain should have the maximum number of allowed ingress and egress
+	// channels.
+	force_clean_hrmp {
+		// ingress channels to a single leaving parachain that need to be closed.
+		let i in 0 .. (<T as configuration::Config>::HrmpMaxInboundChannelsBound::get() - 1);
+		// egress channels to a single leaving parachain that need to be closed.
+		let e in 0 .. (<T as configuration::Config>::HrmpMaxOutboundChannelsBound::get() - 1);
+
+		// we use 10_000 in this benchmark as the prefix of accounts. The components must be greater
+		// than the configured bounds to keep accounts sane.
+		assert!(<T as configuration::Config>::HrmpMaxInboundChannelsBound::get() < PREFIX_0);
+		assert!(<T as configuration::Config>::HrmpMaxOutboundChannelsBound::get() < PREFIX_0);
+
+		// first, update the configs to support this many open channels...
+		assert_ok!(
+			Configuration::<T>::set_hrmp_max_parachain_outbound_channels(frame_system::RawOrigin::Root.into(), e + 1)
+		);
+		assert_ok!(
+			Configuration::<T>::set_hrmp_max_parachain_inbound_channels(frame_system::RawOrigin::Root.into(), i + 1)
+		);
+		// .. and enact it.
+		Configuration::<T>::initializer_on_new_session(&Shared::<T>::scheduled_session());
+
+		let config = Configuration::<T>::config();
+		let deposit: BalanceOf<T> = config.hrmp_sender_deposit.unique_saturated_into();
+
+		let para: ParaId = 1u32.into();
+		let para_origin: crate::Origin = 1u32.into();
+		register_parachain_with_balance::<T>(para, deposit);
+		T::Currency::make_free_balance_be(&para.into_account(), deposit * 256u32.into());
+
+		for ingress_para_id in 0..i {
+			// establish ingress channels to `para`.
+			let ingress_para_id = ingress_para_id + PREFIX_0;
+			let _ = establish_para_connection::<T>(ingress_para_id, 1, ParachainSetupStep::Established);
```

Comment:
`para`?

Reply:
`let para = establish_para_connection::<T>(ingress_para_id, 1, ParachainSetupStep::Established);`? that won't be accurate, because it returns the para id and para origin of both ends of the channel, and I rather just stick to `_` since it is unused.